src/librustc/ty/layout.rs

```diff
@@ -511,41 +512,76 @@ pub struct Struct {
     /// If true, the size is exact, otherwise it's only a lower bound.
     pub sized: bool,
 
-    /// Offsets for the first byte of each field.
+    /// Offsets for the first byte of each field, ordered to match the tys.
+    /// This vector does not go in increasing order.
     /// FIXME(eddyb) use small vector optimization for the common case.
     pub offsets: Vec<Size>,
 
+    /// Maps field indices to GEP indices, depending how fields were permuted.
```

Comment:
I ended up calling these source order and memory order, I don't think there's anything better.  The `gep_index` vec is now `memory_index`.

If there's better nomenclature, I don't know what it is.